packages/rainbowkit/src/components/RainbowKitProvider/RainbowKitProvider.tsx

```diff
@@ -72,7 +72,7 @@ export function RainbowKitProvider({
     [chains]
   );
 
-  useOnDisconnected(clearWalletConnectDeepLink);
+  useAccount({ onDisconnect: clearWalletConnectDeepLink });
```

Comment:
FYI – `useAccount` shouldn't rerender on account changes here as it uses ["tracked values" internally](https://github.com/tmm/wagmi/blob/main/packages/react/src/hooks/utils/useSyncExternalStoreWithTracked.ts).